src/main/java/com/gregtechceu/gtceu/api/recipe/content/Content.java

```diff
@@ -130,15 +131,24 @@ public void drawFluidAmount(GuiGraphics graphics, float x, float y, int width, i
             graphics.pose().pushPose();
             graphics.pose().translate(0, 0, 400);
             graphics.pose().scale(0.5f, 0.5f, 1);
-            int amount = ingredient.getAmount();
             Font fontRenderer = Minecraft.getInstance().font;
+            int amount = ingredient.getAmount();
             String s = FormattingUtil.formatBuckets(amount);
+            int color = 0xFFFFFF;
+            if (content instanceof IntProviderFluidIngredient provider){
+                color = 0xEE0000;
+                amount = provider.getCountProvider().getMaxValue();
+                s = "[" + FormattingUtil.formatBuckets(amount) + "]";
+                if (s.length() > 5) {
+                    s = "X-Y";
+                }
+            }
             if (fontRenderer.width(s) > 32)
                 s = FormattingUtil.formatNumberReadable(amount, true, FormattingUtil.DECIMAL_FORMAT_1F, "B");
             if (fontRenderer.width(s) > 32)
                 s = FormattingUtil.formatNumberReadable(amount, true, FormattingUtil.DECIMAL_FORMAT_0F, "B");
```

Comment:
The color should only be set to red if the output range can't be displayed. Also, your formatting is very different from the item version for some reason?
```suggestion
                int min = ingredient.getCountProvider().getMinValue();
                int max = ingredient.getCountProvider().getMaxValue();
                s = String.format("%s-%s", min, max);
                if (s.length() > 5) {
                    s = "X-Y";
                    color = 0xEE0000;
                }
            } else {
                if (fontRenderer.width(s) > 32)
                    s = FormattingUtil.formatNumberReadable(amount, true, FormattingUtil.DECIMAL_FORMAT_1F, "B");
                if (fontRenderer.width(s) > 32)
                    s = FormattingUtil.formatNumberReadable(amount, true, FormattingUtil.DECIMAL_FORMAT_0F, "B");
            }
```

Reply:
"if the output range can't be displayed" on items a 5 character limit is fine because a single item stack can only be XX-YY; but with Fluids being much higher digit counts, any level of showing range that isn't single-digit whole buckets on both sides will hit the limit and red X-Y out anyway

So I chose to try a different approach: [MAX] for short string quantities, X-Y for if even that little doesn't fit, which it almost always didn't. Honestly it's probably best to just do this as X-Y always because the cases when it _does_ fit within the 5 character limit will be so few and far between.


(Relatedly, when you asked me about the item formatting thing back in the _first_ PR, the reason for the original color change was that the red was hard for me to see and I wanted a better marker that "this field has a ranged value to it"; and a yellow warning felt more appropriate to that than red because red looks more like a "this recipe is broken" error message than a "this recipe has more information to it")

Reply:
yeah, fair enough.
just have it always be `X-Y`. But do keep the red for consistency.